proposals/NNNN-shader-execution-reordering.md

```diff
@@ -0,0 +1,1388 @@
+XXX + 11 | HitObject_RayTMin | Returns the TMin value set in the HitObject.
+XXX + 12 | HitObject_RayTCurrent | Returns the current T value set in the HitObject.
+XXX + 13 | HitObject_WorldRayOrigin | Returns the ray origin in world space.
+XXX + 14 | HitObject_WorldRayDirection | Returns the ray direction in world space.
+XXX + 15 | HitObject_ObjectRayOrigin | Returns the ray origin in object space.
+XXX + 16 | HitObject_ObjectRayDirection | Returns the ray direction in object space.
+XXX + 17 | HitObject_ObjectToWorld3x4 | Returns the object to world space transformation matrix in 3x4 form.
+XXX + 18 | HitObject_ObjectToWorld4x3 | Returns the object to world space transformation matrix in 4x3 form.
+XXX + 19 | HitObject_WorldToObject3x4 | Returns the world to object space transformation matrix in 3x4 form.
+XXX + 20 | HitObject_WorldToObject4x3 | Returns the world to object space transformation matrix in 4x3 form.
+XXX + 21 | HitObject_GeometryIndex | Returns the geometry index committed on hit.
+XXX + 22 | HitObject_InstanceIndex | Returns the instance index committed on hit.
+XXX + 23 | HitObject_InstanceID | Returns the instance id committed on hit.
+XXX + 24 | HitObject_PrimitiveIndex | Returns the primitive index committed on hit.
+XXX + 25 | HitObject_HitKind | Returns the HitKind of the hit.
+XXX + 26 | HitObject_ShaderTableIndex | Returns the shader table index set for this HitObject.
+XXX + 27 | HitObject_SetShaderTableIndex | Returns a HitObject with updated shader table index.
+XXX + 28 | HitObject_LoadLocalRootTableConstant | Returns the root table constant for this HitObject and offset.
+XXX + 29 | HitObject_Attributes | Returns the attributes set for this HitObject.
+
+#### HitObject_TraceRay
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_TraceRay.PayloadT(
+    i32,              ; opcode
+    %dx.types.Handle, ; acceleration structure handle
+    i32,              ; ray flags
+    i32,              ; instance inclusion mask
+    i32,              ; ray contribution to hit group index
+    i32,              ; multiplier for geometry contribution to hit group index
+    i32,              ; miss shader index
+    float,            ; ray origin x
+    float,            ; ray origin y
+    float,            ; ray origin z
+    float,            : tMin
+    float,            ; ray direction x
+    float,            ; ray direction y
+    float,            ; ray direction z
+    float,            ; tMax
+    PayloadT*)        ; payload
+    nounwind
+```
+
+#### HitObject_FromRayQuery
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_FromRayQuery(
+    i32,                           ; opcode
+    %dx.types.RayQueryHandle)      ; ray query
+    nounwind argmemonly
+
+```
+This is used for the HLSL overload of `HitObject::FromRayQuery` that only takes `RayQuery`.
+
+#### HitObject_FromRayQueryWithAttrs
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_FromRayQuery.AttrT(
```

Comment:
This seems meant to match `HitObject_FromRayQueryWithAttrs` in the table above? The names don't match.

Reply:
The idea is that the intrinsic name will have the actual attribute type appended instead.